analyzer/src/types/engine.rs

```diff
@@ -58,6 +59,9 @@ pub struct Chunk {
 
     /// The return type of the chunk.
     pub return_type: TypeId,
+
+    /// Segment of declaration
+    pub declaration_segment: Option<SourceSegment>,
```

Comment:
Isn't the chunk the whole segment? We might actually need to differentiate string sources from environment sources, as a single string can create multiple environments.